Lib/xml/etree/ElementTree.py

```diff
@@ -1726,6 +1731,15 @@ def close(self):
             del self.parser, self._parser
             del self.target, self._target
 
+    def flush(self):
+        self.parser.SetReparseDeferralEnabled(False)
+        try:
+            self.parser.Parse(b"", False)
+        except self._error as v:
+            self._raiseerror(v)
+        finally:
+            self.parser.SetReparseDeferralEnabled(True)
```

Comment:
It always sets the flag to True. Could it restore the old value of the flag?

Reply:
@serhiy-storchaka there is no getter for the old value. What I could have done is only call `SetReparseDeferralEnabled` again to restore if the first call succeeded, but that seemed (and still seems) like a micro-optimization without clear value to me. What do you think?
Is there some serious concern or most curiosity?

Reply:
What is the return value of `SetReparseDeferralEnabled()`? Is it the old value of the flag or the opration success flag? If the latter, the Python wrapper should check it and raise an exception if it fails.

If there is no `GetReparseDeferralEnabled()`, we can simulate it by adding an internal flag in the Python wrapper object around Expact parser, set it in `SetReparseDeferralEnabled()` and add the `GetReparseDeferralEnabled()` method to read it.

Reply:
> What is the return value of `SetReparseDeferralEnabled()`? Is it the old value of the flag or the opration success flag? If the latter, the Python wrapper should check it and raise an exception if it fails.

Operational success. Because of how we use it's guaranteed to succeed, see:

https://github.com/libexpat/libexpat/blob/b7e1a1101133725c79e2170b0870cca8543e5323/expat/lib/xmlparse.c#L2615-L2621

We would only gain complexity with exception code here.

> If there is no `GetReparseDeferralEnabled()`, we can simulate it by adding an internal flag in the Python wrapper object around Expact parser, set it in `SetReparseDeferralEnabled()` and add the `GetReparseDeferralEnabled()` method to read it.

Interesting idea. Could you be explicit in what use case we would add that complexity for?

Reply:
If somebody use `SetReparseDeferralEnabled(False)` to restore the pre-2.6.0 behavior, they cannot use `flush()`, because it sets the flag to True. Now imagine that you need to use a mix of expat 2.6.0 aware and unaware code. To make the latter working you may need to call `SetReparseDeferralEnabled(False)`, but `flush()` cancels it.

Reply:
@serhiy-storchaka I'm not sure how likely that is but I'm afraid you're right about the mix case :+1:

Reply:
@serhiy-storchaka okay, done, pushed.

Reply:
> If there is no `GetReparseDeferralEnabled()`, we can simulate it by adding an internal flag in the Python wrapper object around Expact parser, set it in `SetReparseDeferralEnabled()` and add the `GetReparseDeferralEnabled()` method to read it.

@serhiy-storchaka implemented (and pushed)